### PR TITLE
#166789875 Create endpoints for article

### PR DESCRIPTION
**Pull request by Emile-Nsengimana**
#### What does this PR do?
- create endpoints to:
   > create, delete an article
   >  display a specific, all article(s)
   > update a specific article

#### Description of Task to be completed?
- A user should be able to **CREATE** an articles and **READ/ACCESS** articles created by other authors. 
- User can **ONLY** **UPDATE** and **DELETE** articles  that they have authored.
#### How should this be manually tested?
- First clone it to your machine:
```
https://github.com/andela/ah-lobos-backend.git
```
- Open it using your favorite IDE, I recommend ([vs code](https://code.visualstudio.com/download))

- Install all necessary node modules
```
npm install
```
- Run
```
npm run dev
```
- Use Postman to test out the following endpoints
### API ENDPOINTS
| API | Methods  | Description  |
| ------- | --- | --- |
| `/api/users` | POST | user signup |
| `/api/login` | POST | user login |
| `/api/articles` | POST | create an article |
| `/api/articles/<slug>` | GET | display a specific article |
| `/api/articles/` | GET | display all articles |
| `/api/articles/<slug>` | PUT | update a specific article |
| `/api/article/<slug>` | DELETE | remove a specific article |

#### Any background context you want to provide?
- Create `.env` file with the help of `example.env`
- You need to be registered to the system with a valid working email to be able to access all system functionality
#### What are the relevant pivotal tracker stories?
[#166789875](https://www.pivotaltracker.com/story/show/166789875)
###### 1. create an article
create an article

response of create an article

###### 2. display a specific article
specific article

###### 3. display all articles
display all articles

###### 4. update a specific article
update article 1

update article

###### 5. delete a specific article
delete an article

###### 1. create article
Screen Shot 2019-07-13 at 22 49 35

###### 2. edit a specific article
Screen Shot 2019-07-13 at 22.53.34
Screen Shot 2019-07-13 at 22 53 34

###### 3. display all articles
Screen Shot 2019-07-13 at 22 48 38

###### 4. display a specific article
Screen Shot 2019-07-13 at 22 48 53

###### 5. delete a specific article
Screen Shot 2019-07-13 at 22 49 15

#### Questions:
N/A

### CODE REVIEW

File: src/db/seeders/20190710155302-Users.js [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+      isVerified: true,
+      hash: bcrypt.hashSync('Password12345', 8),
+      createdAt: moment.utc().format(),
+      updatedAt: moment.utc().format()
+    }
+  ], {});
+  },
+
+  down: (queryInterface, Sequelize) => {
+    // return queryInterface.bulkDelete('Users', [{
```
Comment (j4l13n) [inline src/db/seeders/20190710155302-Users.js]: - remove commented lines seeders folder

File: src/index.js [hunk 2]
```diff
@@ -11,7 +11,7 @@ import deleteBlacklist from './helpers/deleteBlacklistTokens';
 
 const app = express(); // setup express application
 
-app.use(logger('dev')); // log requests to the console
+// app.use(logger('dev')); // log requests to the console
```
Comment (j4l13n) [inline src/index.js]: - you have commented the logger in index.js why?

Comment (Emile-Nsengimana) [reply src/index.js]: Sorry for that, I forgot to uncomment it, I wanted to escape logs when running tests

**Review by j4l13n — APPROVED** [review]

**Review by manzif — APPROVED** [review]

File: tests/articlesTest.js [hunk 3]
```diff
@@ -0,0 +1,152 @@
+import Chai from 'chai';
```
Comment (emukungu) [inline tests/articlesTest.js]: change  'Chai' to lowercase

**Review by emukungu — CHANGES_REQUESTED** [review]

File: src/db/config/config.js [hunk 4]
```diff
@@ -4,7 +4,7 @@ const config = {
   development: {
     username: envConfig.db_username,
     password: envConfig.db_password,
-    database: 'authorsHavenDb',
+    database: 'authorshavendb',
```
Comment (emukungu) [inline src/db/config/config.js]: what's the reason for changing the db name?

**Review by emukungu — CHANGES_REQUESTED** [review]

File: src/db/config/config.js (hunk 4 above)
Comment (Emile-Nsengimana) [reply src/db/config/config.js]: sorry for that I somehow forgot to change it back while testing something else on my local machine due to the previous rebase

**Comment by emukungu** [comment]
squash your commits to only one

**Pull request merged**